### PR TITLE
Topology surface entities

### PR DESCRIPTION
**Pull request by chrisrichardson**
Create a list of `bool` for entities which are on the surface of a mesh.

### CODE REVIEW

File: cpp/dolfin/mesh/Topology.cpp [hunk 1]
```diff
@@ -135,6 +135,54 @@ const std::vector<std::int32_t>& Topology::cell_owner() const
...
+{
+  const int tdim = this->dim();
+
+  if (dim >= tdim or dim < 0)
+  {
+    throw std::runtime_error("Invalid entity dimension: "
+                             + std::to_string(dim));
+  }
+
+  std::shared_ptr<const Connectivity> connectivity_facet_cell
```
Comment (garth-wells) [inline cpp/dolfin/mesh/Topology.cpp]: There should be a check that `connectivity_facet_cell` is not null. At least an assertion, but probably an exception should be thrown.

File: cpp/dolfin/mesh/Topology.cpp [hunk 2]
```diff
@@ -135,6 +135,54 @@ const std::vector<std::int32_t>& Topology::cell_owner() const
...
+    for (int i = 0; i < num_facets; ++i)
+    {
+      if (connectivity_facet_cell->size_global(i) == 1)
+        marker[i] = true;
+    }
+    return marker;
+  }
+
+  // Get connectivity from facet to entities of interest (vertices or edges)
+  std::shared_ptr<const Connectivity> connectivity_facet_entity
```
Comment (garth-wells) [inline cpp/dolfin/mesh/Topology.cpp]: This connectivity has already been fetched.

Comment (chrisrichardson) [reply cpp/dolfin/mesh/Topology.cpp]: It hasn't, but it looks very similar. Previously it was `(tdim - 1, tdim)` and nw it is `(tdim - 1, dim)`.

File: cpp/dolfin/mesh/Topology.cpp (hunk 1 above)
Comment (chrisrichardson) [reply cpp/dolfin/mesh/Topology.cpp]: :+1:

**Pull request merged**